my-app/src/app/shared/directives/fresh-indicator.directive.ts

```diff
@@ -0,0 +1,26 @@
+import { Directive, ElementRef, HostBinding, Input, OnChanges, OnInit } from '@angular/core';
+
+@Directive({
+  selector: '[cpFreshIndicator]'
+})
+export class FreshIndicatorDirective implements OnInit {
+  private readonly maxNumberOfDaysMs = 1000 * 60 * 60 * 24 * 14;
+  @Input() creationDate!: Date;
+
+  @HostBinding('style.border') border!: string;
```

Comment:
[FYI] another way is to use `Renderer2` service to update styles on host element